datalab/mlalpha/_summary.py

```diff
@@ -1,107 +1,127 @@
-# Copyright 2016 Google Inc. All rights reserved.
-#
-# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
-# in compliance with the License. You may obtain a copy of the License at
-#
-# http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software distributed under the License
-# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
-# or implied. See the License for the specific language governing permissions and limitations under
-# the License.
-
-"""Implements Cloud ML Summary wrapper."""
-
 import datetime
+import fnmatch
 import glob
+import google.cloud.ml as ml
+import matplotlib.pyplot as plt
 import os
+import pandas as pd
 from tensorflow.core.util import event_pb2
 from tensorflow.python.lib.io import tf_record
 
-import datalab.storage as storage
-
 
 class Summary(object):
-  """Represents TensorFlow summary events from files under a directory."""
+  """Represents TensorFlow summary events from files under specified directories."""
 
-  def __init__(self, path):
+  def __init__(self, paths):
     """Initializes an instance of a Summary.
 
     Args:
-      path: the path of the directory which holds TensorFlow events files.
-            Can be local path or GCS path.
+      path: a list of paths to directories which hold TensorFlow events files.
+            Can be local path or GCS paths. Wild cards allowed.
     """
-    self._path = path
-
-  def _get_events_files(self):
-    if self._path.startswith('gs://'):
-      storage._api.Api.verify_permitted_to_read(self._path)
-      bucket, prefix = storage._bucket.parse_name(self._path)
-      items = storage.Items(bucket, prefix, None)
-      filtered_list = [item.uri for item in items if os.path.basename(item.uri).find('tfevents')]
-      return filtered_list
-    else:
-      path_pattern = os.path.join(self._path, '*tfevents*')
-      return glob.glob(path_pattern)
+    self._paths = [paths] if isinstance(paths, basestring) else paths
+
+  def _glob_events_files(self, paths):
+    event_files = []
+    for path in paths:
+      if path.startswith('gs://'):
+        event_files += ml.util._file.glob_files(os.path.join(path, '*.tfevents.*'))
```

Comment:
Thanks for the pointer. Indeed.
We have many references on google.cloud.ml.util._file.*, which need to be replaced all. I will tackle all in once.